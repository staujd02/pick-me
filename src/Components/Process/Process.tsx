import React from 'react';

export type ProcessProps = {
    steps: React.ReactNode[],
    activeStep: number
}
export type ProcessState = {}

class Process extends React.Component<ProcessProps, ProcessState> {
    render() {
        let activeStep = this.props.activeStep;
        return (
            <React.Fragment>
                {this.props.steps.filter((c, idx) => activeStep === idx)}
            </React.Fragment>
        );
    }
}

export default Process;