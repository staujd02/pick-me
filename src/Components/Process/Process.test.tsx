import React from 'react';
import { ShallowWrapper, shallow, mount } from 'enzyme';
import { formatHTML } from '../../TestUtilities/htmlFormatter';
import Process, { ProcessProps, ProcessState } from './Process';

describe('The Process Component', () => {

    let wrapper: ShallowWrapper<ProcessProps, ProcessState, Process>;

    beforeEach(() => {
        createTasks();
        wrapper = shallow(<Process activeStep={0} steps={StepNodes} />);
    });

    it('renders correctly', () => expect(formatHTML(wrapper.html())).toMatchSnapshot());

    it('shows the first element in the process', () => {
        expect(wrapper.contains(StepNodes[0])).toEqual(true);
    });

    describe('when the user has finished with step 1', () => {

        beforeEach(() => {
            wrapper.setProps({
                activeStep: 1
            });
        });

        it('no longer displays step 1', () => {
            expect(wrapper.contains(StepNodes[0])).toEqual(false);
        });

        it('displays step 2', () => {
            expect(wrapper.contains(StepNodes[1])).toEqual(true);
        });

        describe('when the user has finished step 2', () => {

            beforeEach(() => {
                wrapper.setProps({
                    activeStep: 2
                });
            });

            it('displays step 3', () => {
                expect(wrapper.contains(StepNodes[2])).toEqual(true);
            });

            describe('given step 3 is the last step', () => {
                describe('when the user has finished step 3', () => {

                    beforeEach(() => {
                        wrapper.setProps({
                            activeStep: 0
                        });
                    });

                    it('displays step 1', () => {
                        expect(wrapper.contains(StepNodes[0])).toEqual(true);
                    });
                });
            });
        });
    });

    let StepNodes: React.ReactElement[];

    function createTasks() {
        StepNodes = [
            <div key={0}>Step One</div>,
            <div key={0}>Step Two</div>,
            <div key={0}>Step Three</div>
        ]
    }
});