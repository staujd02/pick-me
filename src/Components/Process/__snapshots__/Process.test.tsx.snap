// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Process Component renders correctly 1`] = `
"
<div>Step One</div>
"
`;
